### PR TITLE
feat: updated claim logic for ledger accounts

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: updated claim logic for ledger accounts

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**Release notes:**

* feat: updated claim logic for ledger accounts

**This pull request makes the following changes:**

**Adds**

* feat: updated claim logic for ledger accounts
image

### CODE REVIEW

**Comment by bobo-k2** [comment]
Hi @impelcrypto can you share an info on how to set up Ledger? Is there a document? Our documentation is empty https://docs.astar.network/docs/integrations/wallets/ledger/ledger-native/

**Comment by impelcrypto** [comment]
@bobo-k2  
Yes, that's the plan :D

**Review by bobo-k2 — APPROVED** [review]
Code looks good to me. I can't test with real device ATM since I just staked some tokens with Ledger device, so I need to wait before be able to claim.

**Comment by impelcrypto** [comment]
@sirius651 @niklabh @Kahonnohak @gluneau Please review this PR when you are available.

File: src/hooks/wallet/useLedger.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    isLedgerNanoS.value = false;
+  };
+
+  const handleLedgerData = async (): Promise<void> => {
+    if (!currentAccount.value || currentAccount.value === null || isH160.value) {
+      handleReset();
+      return;
+    }
+    try {
+      const ledger = new Ledger('hid', 'astar');
```
Comment (impelcrypto) [inline src/hooks/wallet/useLedger.ts:31]: I got this popup when I **haven't** connected Ledger to my laptop (which is annoying for users). I'm not sure how to remove it but I'll try to find a solution...

image

Comment (impelcrypto) [reply src/hooks/wallet/useLedger.ts]: Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/765/commits/b04843f5b4a0e055e5ed7006b09e19b3dfdb0076) commit

**Comment by Kahonnohak** [comment]
Am I supposed to be able to xcm with this stg? Looks I cannot
Screenshot 2023-05-11 at 14 26 32

**Comment by bobo-k2** [comment]
I see an issue when come to dApp staking page from the portal navigation link. Total staked is never shown and estimated reward is 0 which is incorrect
image

If I hit a browser refresh button I got correct info
image

**Comment by impelcrypto** [comment]
@bobo-k2 Have you checked logs in console?

**Comment by bobo-k2** [comment]
> @bobo-k2 Have you checked logs in console?

Yes, nothing is written there.

**Comment by impelcrypto** [comment]
@Kahonnohak 
> Am I supposed to be able to xcm with this stg? Looks I cannot

No, XCM deposit and withdrawal are not supported by Ledger accounts

**Comment by impelcrypto** [comment]
@bobo-k2 What endpoint did you connect to?

**Review by Kahonnohak — COMMENTED** [review]
Lgtm

**Comment by bobo-k2** [comment]
> @bobo-k2 What endpoint did you connect to?

It is Astar endpoint. I don't think this is an endpoint issue, because on page refresh I got correct info and endpoint stays the same.

**Comment by impelcrypto** [comment]
@bobo-k2 
hmm... it looks OK to me. 
https://www.loom.com/share/9a54cbcfb4904687a250f942f7a102c1

**Comment by bobo-k2** [comment]
I see, it might be an isolated case. I also noticed that the issue is not account related. Data doesn't show at first for any wallet (no matter if it is Ledger or not). Production page works fine for me.

**Comment by impelcrypto** [comment]
@Kahonnohak How about you?

**Comment by Kahonnohak** [comment]
Mine works great! 
https://gyazo.com/c474d3a6d1778b0c277cfd861be28682

**Comment by Kahonnohak** [comment]
@bobo-k2 clear storage? we now have one single click to empty the cache:)

**Review by Kahonnohak — APPROVED** [review]

**Review by niklabh — APPROVED** [review]

**Pull request merged**